### DISCUSSION TITLE
How to change IKE_SA destination port

### DISCUSSION THREAD

**Discussion by kldnstn** [strongswan/strongswan#1995] [Q&A] [answered]
Client: 192.168.167.16
Server: 192.168.167.157

Hi, I want my client to reach to the server and establish IPSec with a custom port. Checked the documents and added specific ports in charon(as below, 601 and 4601), but these only changes the source port of the client, not the destination port. Is it possible to change the dest. as well?

 charon {
  install_routes = no
  send_vendor_id = yes
  dns1 = 8.8.8.8
  dns2 = 8.8.4.4
  plugins {
    eap-dynamic {
      preferred = mschapv2, tls, md5
    }
    dhcp {
      identity_lease = yes
    }
  }
  port = 601
  port_nat_t = 4601
.
.
.

│ 07[KNL] using 192.168.167.16 as address to reach 192.168.167.157/32                                                               │
│ 07[IKE] queueing IKE_VENDOR task                                                                                                  │
│ 07[IKE] queueing IKE_INIT task                                                                                                    │
│ 07[IKE] queueing IKE_NATD task                                                                                                    │
│ 07[IKE] queueing IKE_CERT_PRE task                                                                                                │
│ 07[IKE] queueing IKE_AUTH task                                                                                                    │
│ 07[IKE] queueing IKE_CERT_POST task                                                                                               │
│ 07[IKE] queueing IKE_CONFIG task                                                                                                  │
│ 07[IKE] queueing IKE_AUTH_LIFETIME task                                                                                           │
│ 07[IKE] queueing IKE_MOBIKE task                                                                                                  │
│ 07[IKE] queueing CHILD_CREATE task                                                                                                │
│ 07[IKE] activating new tasks                                                                                                      │
│ 07[IKE]   activating IKE_VENDOR task                                                                                              │
│ 07[IKE]   activating IKE_INIT task                                                                                                │
│ 07[IKE]   activating IKE_NATD task                                                                                                │
│ 07[IKE]   activating IKE_CERT_PRE task                                                                                            │
│ 07[IKE]   activating IKE_AUTH task                                                                                                │
│ 07[IKE]   activating IKE_CERT_POST task                                                                                           │
│ 07[IKE]   activating IKE_CONFIG task                                                                                              │
│ 07[IKE]   activating CHILD_CREATE task                                                                                            │
│ 07[IKE]   activating IKE_AUTH_LIFETIME task                                                                                       │
│ 07[IKE]   activating IKE_MOBIKE task                                                                                              │
│ 07[IKE] sending strongSwan vendor ID                                                                                              │
│ 07[IKE] initiating IKE_SA ric[1] to 192.168.167.157                                                                               │
│ 07[IKE] IKE_SA ric[1] state change: CREATED => CONNECTING                                                                         │
│ 07[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) V ]            │
│ 07[NET] sending packet: from 192.168.167.16[601] to 192.168.167.157[500] (1220 bytes)                                             │
│ 14[IKE] retransmit 1 of request with message ID 0                                                                                 │
│ 14[NET] sending packet: from 192.168.167.16[601] to 192.168.167.157[500] (1220 bytes)                                             │
│ 11[IKE] retransmit 2 of request with message ID 0                                                                                 │
│ 11[NET] sending packet: from 192.168.167.16[601] to 192.168.167.157[500] (1220 bytes)                                             │
│ 08[IKE] retransmit 3 of request with message ID 0                                                                                 │
│ 08[NET] sending packet: from 192.168.167.16[601] to 192.168.167.157[500] (1220 bytes)

Status of IKE charon daemon (strongSwan 5.9.5, Linux 5.4.0-162-generic, x86_64):

**Comment by tobiasbrunner** [comment]
Just configure the remote port in the config, i.e. `<conn>.remote_port` in [swanctl.conf](https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html) (has to be the NAT-T port of the remote, see [the docs on custom ports](https://docs.strongswan.org/docs/5.9/features/natTraversal.html#_custom_server_ports) for details).

**Reply by kldnstn** [reply]
Thanks for quick response but it didn't work. Still tries to send to port 500. All edits I've made are like these: Is there anything wrong?



added in strongswan.conf:
charon {
...
  port = 601
  port_nat_t = 4601
... }



uncommented and changed  in charon.conf:
charon {
...
    port = 601
    port_nat_t = 4601
... }




uncommented and changed  in swanctl.conf:
connections {
    <conn> {
...
        remote_port = 4601
... } }

**Reply by tobiasbrunner** [reply]
> Is there anything wrong?

Obviously. Couldn't tell you what without complete configs and/or status output and logs (see #196).

**Reply by kldnstn** [reply]
Okay, adding all necessary logs. There are so many. Hope didn't miss an important one.

[charon_debug.log](https://github.com/strongswan/strongswan/files/13350382/charon_debug.log)
[ip_address.txt](https://github.com/strongswan/strongswan/files/13350387/ip_address.txt)
[ip_route_show_all.txt](https://github.com/strongswan/strongswan/files/13350388/ip_route_show_all.txt)
[ip6tables_save.txt](https://github.com/strongswan/strongswan/files/13350390/ip6tables_save.txt)
[iptables_save.txt](https://github.com/strongswan/strongswan/files/13350393/iptables_save.txt)
[status_of_daemon.txt](https://github.com/strongswan/strongswan/files/13350394/status_of_daemon.txt)
[strongswan_logs.txt](https://github.com/strongswan/strongswan/files/13350396/strongswan_logs.txt)
[charon_conf.txt](https://github.com/strongswan/strongswan/files/13350457/charon_conf.txt)
[ipsec_conf.txt](https://github.com/strongswan/strongswan/files/13350458/ipsec_conf.txt)
[strongswan_conf.txt](https://github.com/strongswan/strongswan/files/13350459/strongswan_conf.txt)
[swanctl_conf.txt](https://github.com/strongswan/strongswan/files/13350460/swanctl_conf.txt)

**Reply by tobiasbrunner** [reply, marked as answer]
OK, you are apparently using the legacy ipsec.conf format. So changing swanctl.conf is completely useless. Just change the corresponding setting in ipsec.conf (`rightikeport`).
